dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java

```diff
@@ -1695,6 +1720,10 @@ public void removeHeader(HeaderKeys key) {
             Tr.debug(tc, "removeHeader(h): " + key.getName());
         }
         removeHdrInstances(findHeader(key), FILTER_YES);
+        int ord = key.getOrdinal();
+        if (inHeadersAdded(ord)) {
```

Comment:
Thanks for finding this problem - I've just pushed a new commit to address the issue found in your review.
As discussed in slack, I took a different tack with the above items, removing the ORD_MAX cap on the headersAdded BitSet, so we no longer need to make any of the above decisions based on the ordinal value.
I also decided to not clear the corresponding bit when a header is removed, since it is possible for ordinals to be recycled - we would rather have a false positive in headersAdded than a false negative. 
And as you suggested, when all headers are removed, we now new up a fresh BitSet for headersAdded, rather than clearing the existing one.